Formula/o/opensearch.rb

```diff
@@ -17,6 +17,12 @@ class Opensearch < Formula
   depends_on "gradle" => :build
   depends_on "openjdk"
 
+  # Support JDK-23 (build time and runtime) --- may be removed in the next release v2.17.2
+  patch do
+    url "https://github.com/opensearch-project/OpenSearch/commit/1e7f6df79c7845ba04ecc4a05979db27965342c7.patch?full_index=1"
+    sha256 "03076625edb55ad0f6f36f8721cb41709fbd732fcb6294f20f1481a6bcc64534"
+  end
+
   def install
```

Comment:
Support JDK-23 (build time and runtime) --- may be removed in the next release v2.17.2